content/blog/improved-kubernetes-await-logic/index.md

```diff
@@ -0,0 +1,293 @@
+
+To that end we've made three improvements to the information Pulumi surfaces when something goes wrong while it's waiting for a resource to become ready (or be deleted):
+
+1. Any Kubernetes Events involving the resource with `type: Warning` are now surfaced by Pulumi as warnings. This can help you see if a Service had trouble provisioning a LoadBalancer, for example.
+
+2. Pulumi will now warn you if it looks like a resource's deletion might be blocked by stuck finalizers. This is a common gotcha that is not always obvious.
+
+3. Error messages from resources that actuate Pods (i.e. Deployments, StatefulSets, etc.) now include status codes and termination messages if their Pods failed to start.
+   A Pod's [termination message](https://kubernetes.io/docs/tasks/debug/debug-application/determine-reason-pod-failure/) can be helpful in `CrashLoopBackOff` situations and is read from `/dev/termination-log` by default.
+
+   (You can configure this path with `terminationMessagePath`, or use `terminationMessagePolicy: FallbackToLogsOnError` to use a Pod's logs as its termination message.)
+
+## Faster deletions with "pulumi.com/deletionPropagationPolicy"
+
+Pulumi uses [foreground](https://kubernetes.io/docs/concepts/architecture/garbage-collection/#foreground-deletion) cascading deletion by default, which deletes a resource _and_ all of its dependents.
+This is the most robust form of deletion but it is also the slowest, and until now there was no way to safely customize this behavior.
+
+The new `pulumi.com/deletionPropagationPolicy` annotation allows you to customize delete behavior while still allowing Pulumi to await readiness when creating or updating the resource.
+
+The annotation accepts three propagation policies:
+
+1. "background": delete the owner resource and leave dependent resources to be asynchronously garbage collected.
+   This is faster than "foreground" deletion propagation, but dependent resources can remain temporarily or even indefinitely if they are not finalized.
+2. "orphan": delete the owner resource and leave dependent resources untouched.
+   This can be useful if you want to keep resources around for migration or debugging purposes.
+3. "foreground": the default behavior of deleting the resource and all of its dependents.
+   This is slower but guarantees all dependents have been cleaned up if it succeeds.
+
+{{% notes type="info" %}}
+Using `skipAwait` to speed up deletion is not recommended when server-side apply is enabled because it is not respected by all resources and it can lead to race conditions during replacement.
+The current behavior is considered buggy, and future releases might change the behavior of the `skipAwait` annotation to have no effect during deletion.
+
+The `pulumi.com/deletionPropagationPolicy` annotation is the recommended way to delete something quickly and safely.
```

Comment:
I don't understand this advice because they're not analogues. You're implying that deletionPropagationPolicy would make the cleanup of the managed resource be quicker, which I suppose would be true if the object has children that are slow to cleanup.

Reply:
Exactly! Changed this to include "...if it has children that are slow to cleanup."